task/buildah-remote/0.1/buildah-remote.yaml

```diff
@@ -0,0 +1,396 @@
+        sed -i 's|^\s*run |RUN . /cachi2/cachi2.env \&\& \\\n    |i' "$dockerfile_path"
+        echo "Prefetched content will be made available"
+      fi
+
+      LABELS=(
+        "--label" "build-date=$(date -u +'%Y-%m-%dT%H:%M:%S')"
+        "--label" "architecture=$(uname -m)"
+        "--label" "vcs-type=git"
+      )
+      [ -n "$COMMIT_SHA" ] && LABELS+=("--label" "vcs-ref=$COMMIT_SHA")
+      [ -n "$IMAGE_EXPIRES_AFTER" ] && LABELS+=("--label" "quay.expires-after=$IMAGE_EXPIRES_AFTER")
+
+      unshare -Uf $UNSHARE_ARGS --keep-caps -r --map-users 1,1,65536 --map-groups 1,1,65536 -- buildah build \
+        $VOLUME_MOUNTS \
+        $BUILDAH_ARGS \
+        ${LABELS[@]} \
+        --tls-verify=$TLSVERIFY --no-cache \
+        --ulimit nofile=4096:4096 \
+        -f "$dockerfile_path" -t $IMAGE $CONTEXT
+
+      container=$(buildah from --pull-never $IMAGE)
+      buildah mount $container | tee /workspace/container_path
+      echo $container > /workspace/container_name
+
+      # Save the SBOM produced by Cachi2 so it can be merged into the final SBOM later
+      if [ -n "${PREFETCH_INPUT}" ]; then
+        cp /tmp/cachi2/output/bom.json ./sbom-cachi2.json
+      fi
+
+      buildah push $IMAGE oci:rhtap-final-image
+      REMOTESSHEOF
+      chmod +x scripts/script-build.sh
+      rsync -ra scripts $SSH_HOST:$BUILD_DIR
+      ssh $SSH_ARGS $SSH_HOST $PORT_FORWARD podman  run $PODMAN_PORT_FORWARD -e BUILDAH_FORMAT=oci -e STORAGE_DRIVER=vfs -e HERMETIC=$(params.HERMETIC) -e PREFETCH_INPUT=$(params.PREFETCH_INPUT) -e CONTEXT=$(params.CONTEXT) -e DOCKERFILE=$(params.DOCKERFILE) -e IMAGE=$(params.IMAGE) -e TLSVERIFY=$(params.TLSVERIFY) -e IMAGE_EXPIRES_AFTER=$(params.IMAGE_EXPIRES_AFTER) -e COMMIT_SHA=$(params.COMMIT_SHA)  --rm  -v $BUILD_DIR/workspaces/source:$(workspaces.source.path):Z  -v $BUILD_DIR/scripts:/script:Z --user=0  $(params.BUILDER_IMAGE)  /script/script-build.sh
+      rsync -ra $SSH_HOST:$BUILD_DIR/workspaces/source/ $(workspaces.source.path)/
+      buildah pull oci:rhtap-final-image
+      buildah images
+      buildah tag localhost/rhtap-final-image $IMAGE
+      container=$(buildah from --pull-never $IMAGE)
+      buildah mount $container | tee /workspace/container_path
+      echo $container > /workspace/container_name
+    securityContext:
+      capabilities:
+        add:
+        - SETFCAP
+    volumeMounts:
+    - mountPath: /var/lib/containers
+      name: varlibcontainers
+    - mountPath: /ssh
+      name: ssh
+      readOnly: true
+    workingDir: $(workspaces.source.path)
+  - image: quay.io/redhat-appstudio/syft:v0.85.0
+    name: sbom-syft-generate
+    resources: {}
+    script: |
+      syft dir:$(workspaces.source.path) --file=$(workspaces.source.path)/sbom-source.json --output=cyclonedx-json
+      find $(cat /workspace/container_path) -xtype l -delete
+      syft dir:$(cat /workspace/container_path) --file=$(workspaces.source.path)/sbom-image.json --output=cyclonedx-json
```

Comment:
Does Syft work properly when inspecting images built for arches other than x86_64?

Reply:
(it's OK to defer this to a follow-up story IMO, let's just make sure we have one if we can't answer this now)

Reply:
Yes, syft has support for reading from different architectures. I know this because the older version we were originally using had a bug in its handling of big endian architectures, and upgrading to a later version fixed it (while investigating the bug I poked through the code and found that it explicitly has handling for different architectures).